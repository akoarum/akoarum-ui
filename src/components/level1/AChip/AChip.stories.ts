import { storiesOf } from '@storybook/vue'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import AChip from './AChip.vue'

storiesOf('level1/AChip', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { AChip },
    props: {
      small: { default: boolean('小', false) },
      texts: { default: text('テキスト', 'テキスト') }
    },
    template: `
      <AChip :small="small">{{ texts }}</AChip>
    `
  }), { info: {} })
  .add('Primary', () => ({
    components: { AChip },
    props: {
      small: { default: boolean('小', false) },
      texts: { default: text('テキスト', 'テキスト') }
    },
    template: `
      <AChip :small="small" primary>{{ texts }}</AChip>
    `
  }), { info: {} })
  .add('Secondary', () => ({
    components: { AChip },
    props: {
      small: { default: boolean('小', false) },
      texts: { default: text('テキスト', 'テキスト') }
    },
    template: `
      <AChip :small="small" secondary>{{ texts }}</AChip>
    `
  }), { info: {} })
  .add('Tertiary', () => ({
    components: { AChip },
    props: {
      small: { default: boolean('小', false) },
      texts: { default: text('テキスト', 'テキスト') }
    },
    template: `
      <AChip :small="small" tertiary>{{ texts }}</AChip>
    `
  }), { info: {} })
  .add('Caution', () => ({
    components: { AChip },
    props: {
      small: { default: boolean('小', false) },
      texts: { default: text('テキスト', 'テキスト') }
    },
    template: `
      <AChip :small="small" caution>{{ texts }}</AChip>
    `
  }), { info: {} })

