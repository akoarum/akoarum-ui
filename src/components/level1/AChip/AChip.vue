<template>
  <i :class="styledClasses" class="aChip"><slot /></i>
</template>

<script lang="ts">
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  small?: boolean
  primary?: boolean
  secondary?: boolean
  tertiary?: boolean
  caution?: boolean
}

export default defineComponent({
  props: {
    small: { type: Boolean },
    primary: { type: Boolean },
    secondary: { type: Boolean },
    tertiary: { type: Boolean },
    caution: { type: Boolean }
  },
  setup(props: Props) {
    const styledClasses = computed<string[]>(() => {
      const classes: string[] = []
      if (props.small) classes.push('-small')
      if (props.primary) classes.push('-primary')
      if (props.secondary) classes.push('-secondary')
      if (props.tertiary) classes.push('-tertiary')
      if (props.caution) classes.push('-caution')
      return classes
    })

    return { styledClasses }
  }
})
</script>

<style lang="scss" scoped>
.aChip {
  display: inline-block;
  border-radius: 3px;
  padding: 5px 8px 4px;
  background: #acaeaf;
  font-size: 12px;
  line-height: 1;
  color: #fff;

  &.-small {
    border-radius: 0;
    padding: 4px 6px 3px;
    font-size: 10px;
    font-weight: bold;
  }
  &.-primary {
    background: $color-primary;
  }
  &.-secondary {
    background: #6fcf97;
  }
  &.-caution {
    background: #e37070;
  }
}
</style>
