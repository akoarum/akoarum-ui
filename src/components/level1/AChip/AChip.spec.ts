import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import AChip from './AChip.vue'

describe('AChip', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(AChip, {
      propsData: {},
      slots: {
        default: 'テキスト'
      },
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('styledClasses: propsによって見た目を制御できる', () => {
      wrapper.setProps({ small: true, tertiary: true })
      expect(wrapper.vm.styledClasses).toEqual(['-small', '-tertiary'])
    })
  })
})
