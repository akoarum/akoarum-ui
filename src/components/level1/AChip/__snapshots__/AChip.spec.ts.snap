// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AChip initialized snapshot 1`] = `
<i
  class="aChip"
>
  テキスト
</i>
`;
